internal/backend/sessions/sessionworkflows/workflow.go

```diff
@@ -478,6 +482,21 @@ func (w *sessionWorkflow) run(wctx workflow.Context) (prints []string, err error
 			kittehs.ContainedIn(argNames...),
 		)
 
+		// If session was triggered by a connection, pass the connection to the entrypoint.
+		var trigger struct{ ConnectionId string }
+		if err := w.data.Session.Inputs()["trigger"].UnwrapInto(&trigger); err == nil && trigger.ConnectionId != "" {
+			if _, conn := kittehs.FindFirst(w.data.Connections, func(c sdktypes.Connection) bool {
+				return c.ID().String() == trigger.ConnectionId
+			}); conn.IsValid() {
+				if kwargs["conn"], err = sdktypes.NewModuleValue(
+					conn.Name(),
+					w.executors.GetValues(conn.ID().String()),
+				); err != nil {
+					return prints, err
+				}
+			}
+		}
```

Comment:
maybe it would be more readable if you convert the if checks and exit faster

less indentations are better, 
here you check that trigger unwraps and has connectionId and find it in data.connections and is valid and conn is kwargs and you getValues by connID 
which is really hard to follow and get what you try to do